ledger/src/shred.rs

```diff
@@ -467,7 +473,25 @@ impl Shred {
         self.common_header.signature
     }
 
-    pub fn seed(&self) -> [u8; 32] {
+    pub fn seed(
+        &self,
+        leader_schedule_cache: &Arc<LeaderScheduleCache>,
+        bank_forks: &Arc<RwLock<BankForks>>,
+    ) -> [u8; 32] {
+        let root_bank = bank_forks.read().unwrap().root_bank();
+        if enable_deterministic_seed(self.slot(), &root_bank) {
+            if let Some(leader_pubkey) =
+                leader_schedule_cache.slot_leader_at(self.slot(), Some(&root_bank))
+            {
+                let h = hashv(&[
+                    &self.slot().to_le_bytes(),
+                    &self.index().to_le_bytes(),
+                    &leader_pubkey.to_bytes(),
+                ]);
```

Comment:
did you get a chance to get a final confirmation from Anatoly re this?

Reply:
yes, I did check with him about using this hashed tuple